crates/sui-adapter/src/adapter.rs

```diff
@@ -80,21 +80,24 @@ pub fn execute<E: Debug, S: ResourceResolver<Error = E> + ModuleResolver<Error =
     let is_genesis = ctx.digest() == TransactionDigest::genesis();
     let TypeCheckSuccess {
         module_id,
-        args,
+        mut args,
         object_data,
         by_value_objects,
         mutable_ref_objects,
+        has_ctx_arg,
     } = resolve_and_type_check(&objects, &module, function, &type_args, args, is_genesis)?;
 
-    let mut args = args;
-    args.push(ctx.to_vec());
+    if has_ctx_arg {
+        args.push(ctx.to_vec());
+    }
     execute_internal(
         vm,
         state_view,
         &module_id,
         function,
         type_args,
         args,
+        has_ctx_arg,
         object_data,
         by_value_objects,
         mutable_ref_objects,
```

Comment:
Should we add ctx to mutable_ref_objects?

Reply:
Not sure how that would work out, because the ctx isn't an object